src/Dfe.PlanTech.Application/Submission/Commands/RecordSubmitAnswerCommands.cs

```diff
@@ -0,0 +1,54 @@
+using Dfe.PlanTech.Application.Response.Interface;
+using Dfe.PlanTech.Application.Submission.Interfaces;
+using Dfe.PlanTech.Domain.Answers.Models;
+using Dfe.PlanTech.Domain.Questions.Models;
+using Dfe.PlanTech.Domain.Responses.Models;
+
+namespace Dfe.PlanTech.Application.Submission.Commands
+{
+    public class RecordSubmitAnswerCommands
+    {
+
+        private readonly IRecordQuestionCommand _recordQuestionCommand;
+        private readonly IRecordAnswerCommand _recordAnswerCommand;
+        private readonly ICreateSubmissionCommand _createSubmissionCommand;
+        private readonly ICreateResponseCommand _createResponseCommand;
+
+        public RecordSubmitAnswerCommands(
+            IRecordQuestionCommand recordQuestionCommand,
+            IRecordAnswerCommand recordAnswerCommand,
+            ICreateSubmissionCommand createSubmissionCommand,
+            ICreateResponseCommand createResponseCommand)
+        {
+            _recordQuestionCommand = recordQuestionCommand;
+            _recordAnswerCommand = recordAnswerCommand;
+            _createSubmissionCommand = createSubmissionCommand;
+            _createResponseCommand = createResponseCommand;
+        }
+
+        public async Task<int> RecordQuestion(RecordQuestionDto recordQuestionDto)
+        {
+            if (recordQuestionDto.QuestionText == null) throw new ArgumentNullException(nameof(recordQuestionDto));
```

Comment:
I don't think `ArgumentNullException` is the most appropriate here, especially since (in the instances in this class) the argument itself isn't null.

Perhaps `NullReferenceException` might be more appropriate?

Reply:
Yeah I had a little internal debate about that, I agree that `NullReferenceException` is the best fit here

Resolved in: [c3c2b9a](https://github.com/DFE-Digital/plan-technology-for-your-school/pull/151/commits/c3c2b9ad56d376f16f4506c7ea7f3c8907928d26)